src/resolvers/deck/createDeck.ts

```diff
@@ -28,3 +35,190 @@ export const createDeck = mutationWithClientMutationId({
     return { deck }
   },
 })
+
+export const importDeck = mutationWithClientMutationId({
+  name: 'ImportDeck',
+  description: 'Import/Duplicate a deck entity',
+  inputFields: {
+    id: { type: GraphQLNonNull(GraphQLID), description: 'Public deck id' },
+  },
+  outputFields: {
+    deck: { type: DeckType, description: 'Imported deck' },
+  },
+  mutateAndGetPayload: async (
+    { id },
+    { user, publishedDeckLoader, modelLoader, fieldsByModelLoader }: Context
+  ) => {
+    const helperMap: {
+      updateModelIndex: { oldId: string; newId: string }[]
+      updateFieldIndex: { oldId: string; newId: string }[]
+    } = {
+      updateModelIndex: [],
+      updateFieldIndex: [],
+    }
+    const { id: deckId } = fromGlobalId(id)
+    const publishedDeck = await publishedDeckLoader.load(deckId)
+    const deckNotes = await NoteModel.find({ deckId: publishedDeck?._id })
+
+    // Find requires models/fields & templates
+    const modelIds = deckNotes.map((note) => note.modelId.toString())
+    const uniqueModelIds = modelIds.filter((v, i) => modelIds.indexOf(v) === i)
+    const models = []
+
+    for await (const modelId of uniqueModelIds) {
+      const model = await modelLoader.load(modelId)
+      const currentModelFields = await fieldsByModelLoader.load(model?._id)
+      const currentModelTemplates = await TemplateModel.find({
+        modelId: model?._id,
+      })
+
+      models.push({
+        model,
+        fields: currentModelFields,
+        templates: currentModelTemplates,
+      })
+    }
+
+    // Create deck
+    const newDeck = await DeckModel.create({
+      title: `${publishedDeck?.title} - new`,
```

Comment:
we should keep the original deck name
```suggestion
      title: publishedDeck?.title,
```

Reply:
ups, this was just an oversight